APITaxi_models/zupc.py

```diff
@@ -63,3 +67,40 @@ def top(self):
     @property
     def right(self):
         return self.bounds[2]
+
+    @staticmethod
+    def is_inactive_period():
+        inactive_filter_period = current_app.config['INACTIVE_FILTER_PERIOD']
+        hour = datetime.now().hour
+        if inactive_filter_period[0] > inactive_filter_period[1]:
+            return hour >= inactive_filter_period[0] or hour <= inactive_filter_period[1]
+        else:
+            return inactive_filter_period[0] <= hour <= inactive_filter_period[1]
+
+    @classmethod
+    def get_max_distance(cls, zupc_customer):
+        #We can deactivate the max radius for a certain zone
+        if cls.is_inactive_period:
+            return current_app.config['DEFAULT_MAX_RADIUS']
+        else:
+            return min([v for v in [v[2] for v in zupc_customer] if v and v>0]
```

Comment:
`[v[2] for v in zupc_customer]` doesn't work. What do you want to do here? Each element of zupc_customer is a ZUPC object, which is not subscriptable.

Reply:
I want to return the minimum positive max_distance.

```suggestion
            return min([z.max_distance for z in zupc_customer if z.max_distance and z.max_distance>0])
```